src/built-in-types/MapType.ts

```diff
@@ -5,27 +5,42 @@ export interface Spec {
   [key: string]: EcsType
 }
 
-export type Result<T extends Spec> = {
+/** Include property keys from T where the property is assignable to U */
+type IncludeUndefined<T> = {
+  [P in keyof T]: undefined extends T[P] ? P : never
+}[keyof T]
+
+/** Excludes property keys from T where the property is assignable to U */
+type ExcludeUndefined<T> = {
+  [P in keyof T]: undefined extends T[P] ? never : P
+}[keyof T]
+
+type OnlyOptionalUndefinedTypes<T> = { [K in IncludeUndefined<T>]?: T[K] }
+type OnlyNonUndefinedTypes<T> = { [K in ExcludeUndefined<T>]: T[K] }
+
+type ToOptional<T> = OnlyOptionalUndefinedTypes<T> & OnlyNonUndefinedTypes<T>
+
+export type Result<T extends Spec> = ToOptional<{
   [K in keyof T]: T[K] extends EcsType
-    ? ReturnType<T[K]['deserialize']>
+    ? ReturnType<T[K]['deserialize']> extends undefined
```

Comment:
i think there is no need for this change. 🤔

Reply:
thx